assets/js/base/components/block-error-boundary/index.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import BlockError from './block-error';
+import './style.scss';
+
+interface DerivedStateReturn {
+	errorMessage: JSX.Element | string;
+	hasError: boolean;
+}
+
+interface Error {
+	status: string;
+	statusText: string;
+	message: string;
+}
+
+type RenderError = ( props: RenderErrorProps ) => React.ReactNode | null;
+
+interface RenderErrorProps {
+	errorMessage: React.ReactNode;
+}
```

Comment:
Super extra nitpick here: I'd define `RenderErrorProps` before `RenderError` so that they are used in order and are easier to read and parse. Not a big deal if you don't want to change it, it's totally a matter of opinion.